packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -247,12 +259,38 @@ class BuildInfo {
     };
   }
 
+
   /// Convert to a structured string encoded structure appropriate for usage as
   /// environment variables or to embed in other scripts.
   ///
   /// Fields that are `null` are excluded from this configuration.
   Map<String, String> toEnvironmentConfig() {
-    return <String, String>{
+    final Map<String, String> result = <String, String>{};
+    if (dartDefineConfigJsonMap != null) {
+      final List<String> systemVars=<String>[
```

Comment:
I still do not understand why you need this new variable `systemVars` if you just compare what the user send on the config file and what you already have on `dartDefineConfigJsonMap` wouldn't be enough?

Reply:
I misunderstood, I thought you let me judge completely if the variable is duplicated. Because the above variable is conditional, if the condition is not satisfied, there is no variable, and I just set the above variable, which will cause coverage